dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -91,6 +92,18 @@ public KernelBuilder WithMemoryStorage(IMemoryStore storage)
         return this;
     }
 
+    /// <summary>
+    /// Add prompt template engine to the kernel to be built.
+    /// </summary>
+    /// <param name="promptTemplateEngine">Prompt template engine to add.</param>
+    /// <returns>Updated kernel builder including the prompt template engine.</returns>
+    public KernelBuilder WithPromptTemplateEngine(IPromptTemplateEngine promptTemplateEngine)
```

Comment:
I think we're going in the opposite direction of the plan to simplify the public API, reducing methods, and the plan to use .NET DI. 

Before merging this branch into main we'll need to address this.

Reply:
Completely agree with you @dluc that the API should be simplified but not by decreasing number of methods but by redesigning SK SDK components so that those methods are moved to the classes they naturally belong to.